src/NUnitEngine/nunit.engine/Services/ExtensionAssembly.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ***********************************************************************
+// Copyright (c) 2016 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.IO;
+using Mono.Cecil;
+
+namespace NUnit.Engine.Services
+{
+    public class ExtensionAssembly
+    {
+        public ExtensionAssembly(string filePath, bool fromWildCard)
+        {
+            FilePath = filePath;
+            FromWildCard = fromWildCard;
+        }
+
+        public string FilePath { get; private set; }
+        public bool FromWildCard { get; private set; }
+
+        private AssemblyDefinition _assemblyDefinition;
+        public AssemblyDefinition Assembly
+        {
+            get
+            {
+                if (_assemblyDefinition == null)
```

Comment:
We may not need it here, but should we get in the habit of using Lazy<T> for all lazy initialization? It may not be needed now, but it is easy to implement and future proofs the code.